src/esp/gfx/replay/Player.cpp

```diff
@@ -101,10 +136,7 @@ void Player::close() {
 
 void Player::clearFrame() {
   for (const auto& pair : createdInstances_) {
-    // TODO: use NodeDeletionHelper to safely delete nodes owned by the Player.
-    // the deletion here is unsafe because a Player may persist beyond the
-    // lifetime of these nodes.
-    delete pair.second;
+    implementation_->deleteAssetInstance(pair.second);
   }
   createdInstances_.clear();
   assetInfos_.clear();
```

Comment:
@eundersander continuing with our conversation, I feel like I should add a `AbstractPlayerImplementation::clear()` interface, which either `delete`s in the SceneGraph or `clear()`s in the batch renderer; and call that from here instead. Does that sound alright?

Reply:
A sketch for this is in e96df3ba9c372473d7e5013564451a3d0f7c2d06.